src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java

```diff
@@ -817,11 +817,22 @@ protected void addInjectables(DeserializationContext ctxt,
             for (Map.Entry<Object, AnnotatedMember> entry : raw.entrySet()) {
                 AnnotatedMember m = entry.getValue();
                 final JacksonInject.Value injectableValue = introspector.findInjectableValue(m);
-                final Boolean optional = injectableValue == null ? null : injectableValue.getOptional();
+                final Boolean optional;
+                final Boolean useInput;
 
-                builder.addInjectable(PropertyName.construct(m.getName()),
-                        m.getType(),
-                        beanDesc.getClassAnnotations(), m, entry.getKey(), optional);
+                if (injectableValue == null) {
+                    optional = null;
+                    useInput = null;
+                } else {
+                    optional = injectableValue.getOptional();
+                    useInput = injectableValue.getUseInput();
+                }
+
+                if (!Boolean.TRUE.equals(useInput)) {
```

Comment:
I think this is slightly off: Javadoc for `useInput` says:

    Default is `OptBoolean.DEFAULT`, which translates to `OptBoolean.TRUE`.

I can change this.

Reply:
Oh. Never mind, later comment spells it out